forc-plugins/forc-debug/src/cli/state.rs

```diff
@@ -0,0 +1,201 @@
+use crate::{cli::commands::Commands, FuelClient};
+use rustyline::{
+    completion::Completer,
+    highlight::{CmdKind, Highlighter},
+    hint::Hinter,
+    validate::{ValidationContext, ValidationResult, Validator},
+    Context, Helper,
+};
+use serde_json::Value;
+use std::{borrow::Cow, fs, path::Path};
+
+pub struct State {
+    pub client: FuelClient,
+    pub session_id: String,
+}
+
+impl State {
+    pub fn new(client: FuelClient) -> Self {
+        Self {
+            client,
+            session_id: String::new(),
+        }
+    }
+}
+
+pub struct DebuggerHelper {
+    pub commands: Commands,
+}
+
+impl DebuggerHelper {
+    pub fn new() -> Self {
+        Self {
+            commands: Commands::new(),
+        }
+    }
+}
+
+impl Completer for DebuggerHelper {
+    type Candidate = String;
+
+    fn complete(
+        &self,
+        line: &str,
+        pos: usize,
+        _ctx: &Context<'_>,
+    ) -> rustyline::Result<(usize, Vec<Self::Candidate>)> {
+        let words: Vec<&str> = line[..pos].split_whitespace().collect();
+        let word_start = line[..pos].rfind(char::is_whitespace).map_or(0, |i| i + 1);
+        let word_to_complete = &line[word_start..pos];
+
+        // Transaction command context
+        if let Some(first_word) = words.first() {
+            if self.commands.is_tx_command(first_word) && line[..word_start].ends_with(' ') {
+                return Ok((word_start, get_transaction_files(word_to_complete)));
+            }
+
+            // Register command context
+            if self.commands.is_register_command(first_word) && line[..word_start].ends_with(' ') {
+                let register_names = vec![
```

Comment:
nit: We should use the constant `names::REGISTERS` here

Reply:
Good catch! added in [a356e73](https://github.com/FuelLabs/sway/pull/6830/commits/a356e73ba873f769f47979dcb5a6685c93f01dee)